### PR TITLE
Update model version integer format error code

### PR DESCRIPTION
**Pull request by andychow-db**
## What changes are proposed in this pull request?

When a user specifies a model version's version, it should be an integer
or a string integer. If not, the current error returns an INTERNAL_ERROR.

This PR converts the error code from INTERNAL_ERROR to
INVALID_PARAMETER_VALUE and updates the message to be more
understandable.

### Unit testing

Updated the client level unit tests for model registry which invoke a sql alchemy backend store. Validated that the behavior results in the updated error code and messaging.

### Manual testing

Ran the server using the command:
```
mlflow server --backend-store-uri sqlite:///mlflow.db --default-artifact-root /Users/Andy.Chow/Workspace/mlflow-tutorial/artifact-root
```

And tested the updated implementation via cURL 

Screen Shot 2020-03-30 at 9 51 23 AM

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

When user supplies an invalid model version number as anything else, we return an INTERNAL_ERROR. This update changes this error code to INVALID_PARAMETER_VALUE and makes the message easier to understand.  

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/store/model_registry/sqlalchemy_store.py [hunk 1]
```diff
@@ -282,8 +282,8 @@ def _get_sql_model_version(cls, session, name, version):
         try:
             version = int(version)
         except ValueError:
-            raise MlflowException("The version for sql alchemy store must be an integer, got '{}'"
-                                  .format(version))
+            raise MlflowException("Model Version's version must be an integer, got '{}'"
```
Comment (stbof) [inline mlflow/store/model_registry/sqlalchemy_store.py]: Model version must be an integer

File: tests/tracking/test_model_registry.py [hunk 2]
```diff
@@ -246,6 +247,19 @@ def test_create_and_query_model_version_flow(mlflow_client, backend_store_uri):
...
+    name = 'GetModelVersionTest'
+    mlflow_client.create_registered_model(name)
+    mlflow_client.create_model_version(name, "path/to/model", "run_id_1")
+    model_version = mlflow_client.get_model_version(name, '1')
+    assert model_version.name == name
+    assert model_version.version == '1'
+
+    with pytest.raises(MlflowException) as ex:
+        mlflow_client.get_model_version(name=name, version="something not correct")
+    assert "INVALID_PARAMETER_VALUE: Model Version's version must be an integer" in str(ex.value)
```
Comment (stbof) [inline tests/tracking/test_model_registry.py]: Model version must be an integer

**Comment by andychow-db** [comment]
I've updated the messaging per @stbof's feedback and reran the tests both manual and unit tests.

File: tests/tracking/test_model_registry.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
 
 from mlflow.entities.model_registry import RegisteredModel
 from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE, ErrorCode
```
Comment (sueann) [inline tests/tracking/test_model_registry.py]: looks like this is failing in lint as unused imports

Comment (andychow-db) [reply tests/tracking/test_model_registry.py]: Yeah, I'm running through the tests again right now. I'll ping you once it's fixed.

**Review by sueann — APPROVED** [review]
lgtm - thanks!

**Pull request merged**